[Bug]: 7.2.0版本无法读取GB2312编码的TXT文件

### 问题描述

7.2.0版本下无法在读取文件之前选择原始语言，会在读取前自动检测编码。但如果小说TXT为GB2312编码，程序会优先检测gb18030编码，不匹配之后使用utf-8编码读取，导致程序无法读取GB2312编码的TXT文件。

在7.0.9版本手动选择语言为中文之后，可以成功读取GB2312编码的TXT文件。

### 报错日志

[ERROR] 翻译项目数据载入失败，请检查项目类型与输入文件夹设置。
'utf-8' codec can't decode byte 0xc9 in position 2: invalid continuation byte
Traceback (most recent call last):
  File "UserInterface\EditView\Startup\StartupPage.py", line 217, in _load_project_worker
  File "ModuleFolders\Domain\FileReader\FileReader.py", line 96, in read_files
  File "ModuleFolders\Domain\FileReader\DirectoryReader.py", line 91, in read_source_directory
  File "ModuleFolders\Domain\FileReader\AutoTypeReader.py", line 123, in read_source_file
  File "ModuleFolders\Domain\FileReader\BaseReader.py", line 59, in read_source_file
  File "ModuleFolders\Domain\FileReader\TxtReader.py", line 31, in on_read_source
  File "pathlib.py", line 1028, in read_text
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc9 in position 2: invalid continuation byte
[WARNING] charset_normalizer检测的编码 gb18030 置信度不足 (coherence: 31.11%, chaos: 0.00%)

### 软件版本

7.2.0

### 接口平台

_No response_

### 接口地址

_No response_

### 接口格式

_No response_

### 模型名称

GalTransl-v4-4B-2601

### 文件类型

TXT

### 原文语言

简体中文

### 译文语言

英文

### 并行任务数

16

### 提示词预设

通用

--- a/ModuleFolders/Domain/FileReader/ReaderUtil.py
+++ b/ModuleFolders/Domain/FileReader/ReaderUtil.py
@@ -236,8 +236,18 @@ def detect_file_encoding(file_path: Union[str, pathlib.Path], min_confidence: fl
             # 验证检测结果：尝试解码
             try:
                 content_bytes.decode(detected_encoding)
-                # 检查置信度和混乱度（chaos < 30%表示质量可接受）
-                if confidence >= min_confidence and chaos_percent < 30.0:
+                # chaos 是 charset_normalizer 设计上的主判据，衡量解码乱码率，越低越好
+                # coherence 衡量文本是否像某种自然语言，但对 CJK 等非拉丁文字天然偏低
+                # 库自身的 chaos 默认阈值为 0.2(20%)，此处与其对齐；coherence 仅作辅助判断
+                if chaos_percent < 5.0:
+                    # 近乎零乱码：多字节编码(如 GBK/GB18030)能满足此条件本身就是强信号
+                    # coherence 不作要求，因为 CJK 文字的字母频率统计天然偏低
+                    return detected_encoding
+                elif chaos_percent < 20.0 and confidence > 0.08:
+                    # 低乱码且有基本语言特征，与库默认阈值对齐
+                    return detected_encoding
+                elif chaos_percent < 30.0 and confidence >= min_confidence:
+                    # 中等乱码，严格要求 coherence 才接受，保留原有逻辑作为兜底
                     return detected_encoding
                 else:
                     rich.print(

